src/main.rs

```diff
@@ -61,69 +59,122 @@ impl BFInterp {
+                        Instruction::URem(urem) => self.int_bin_operations(
+                            &urem.operand0,
+                            &urem.operand1,
+                            &urem.dest,
+                            BinOps::Rem,
+                        ),
+                        Instruction::SRem(srem) => self.int_bin_operations(
+                            &srem.operand0,
+                            &srem.operand1,
+                            &srem.dest,
+                            BinOps::Rem,
+                        ),
+                        Instruction::FSub(fsub) => self.fl_bin_operations(
+                            &fsub.operand0,
+                            &fsub.operand1,
+                            &fsub.dest,
+                            BinOps::Sub,
+                        ),
+                        Instruction::SExt(sext) => {
+                            self.szext(&sext.operand, &sext.to_type, &sext.dest)
+                        }
+                        Instruction::FMul(fmul) => self.fl_bin_operations(
+                            &fmul.operand0,
+                            &fmul.operand1,
+                            &fmul.dest,
+                            BinOps::Mul,
+                        ),
+                        Instruction::FDiv(fdiv) => self.fl_bin_operations(
+                            &fdiv.operand0,
+                            &fdiv.operand1,
+                            &fdiv.dest,
+                            BinOps::Div,
+                        ),
+                        Instruction::ICmp(icmp) => {
+                            self.icmp(icmp.predicate, &icmp.operand0, &icmp.operand1, &icmp.dest)
+                        }
+                        Instruction::ZExt(zext) => {
+                            self.szext(&zext.operand, &zext.to_type, &zext.dest)
+                        }
+                        _ => todo!(),
+                    }
                 }
-                Instruction::FDiv(fdiv) => {
-                    self.fl_bin_operations(&fdiv.operand0, &fdiv.operand1, &fdiv.dest, BinOps::Div)
+
+                match &bb.term {
+                    Terminator::Ret(ret) => {
+                        match ret.return_operand.as_ref() {
+                            Some(op) => match op {
+                                LocalOperand { .. } => todo!(),
+                                ConstantOperand(con_ref) => match con_ref.as_ref() {
+                                    Constant::Int { bits: _, .. } => {
+                                        // return Some(*value);
+                                    }
+                                    _ => todo!(),
+                                },
+                                MetadataOperand => todo!(),
+                            },
+                            None => {}
+                        }
+                    }
+                    Terminator::Br(br) => self.it_bb(func, br.dest.clone()),
```

Comment:
We need to be careful here, since we are now recursively calling `it_bb` and we might execute so many branch instructions that we run out of memory (https://en.wikipedia.org/wiki/Call_stack). We should turn this into an iterative loop instead.